extensions/echo-message-3.3.md

```diff
@@ -0,0 +1,63 @@
+echo-message client capability specification
+--------------------------------------------
+
+Copyright (c) 2014 Attila Molnar <attilamolnar@hush.com>
+Copyright (c) 2014 J-P Nurmi <jpnurmi@gmail.com>
+
+Unlimited redistribution and modification of this document is allowed
+provided that the above copyright notice and this permission notice
+remains in tact.
+
+## Description
+
+This client capability MUST be named `echo-message`.
+
+If enabled, servers MUST send `PRIVMSG` and `NOTICE` messages back to
+the client that sent them. If servers apply any modifications to these
+messages, they MUST send the final version of the message back to the
+originating client.
+
+For clients, receiving a message with themselves as the sender acts as
+an acknowledgement that the message has been delivered to the server.
+Clients that receive self-sent `PRIVMSG` and `NOTICE` messages, MUST
+treat them the same way as if the client itself would have sent the
+message to the target. Clients may choose to disable local echoing
+of sent `PRIVMSG` and `NOTICE` messages altogether, or present them
+in pending state.
+
+## Use cases
+
+The capability is useful for clients to get an acknowledgement that a
+message was delivered. It can be also used for measuring lag between
+client and server, that is, the time it takes from sending a message
+to receiving it back.
+
+Furthermore, the capability is useful for users that have multiple
+clients attached to a bouncer. In this scenario, when users send
+messages from one client, the messages get automatically relayed to
+other attached clients. This allows all attached clients to display
+full conversation.
```

Comment:
This is apparently a more flexible version of `self-message`.  It should replace `self-message` in 3.2, I created a clarification request (#135) allowing that to be implemented using the argument that `self-message` presence would be an errata since it is to be replaced in 3.3 with this one...